pkg/util/validation/limits.go

```diff
@@ -281,7 +281,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&l.MaxFetchedChunkBytesPerQuery, MaxChunkBytesPerQueryFlag, 0, "The maximum size of all chunks in bytes that a query can fetch from ingesters and store-gateways. This limit is enforced in the querier and ruler. 0 to disable.")
 	f.Uint64Var(&l.MaxEstimatedMemoryConsumptionPerQuery, MaxEstimatedMemoryConsumptionPerQueryFlag, 0, "The maximum estimated memory a single query can consume at once, in bytes. This limit is only enforced when Mimir's query engine is in use. This limit is enforced in the querier. 0 to disable.")
 	f.Var(&l.MaxPartialQueryLength, MaxPartialQueryLengthFlag, "Limit the time range for partial queries at the querier level.")
-	f.Var(&l.MaxQueryLookback, "querier.max-query-lookback", "Limit how long back data (series and metadata) can be queried, up until <lookback> duration ago. This limit is enforced in the query-frontend, querier and ruler. If the requested time range is outside the allowed range, the request will not fail but will be manipulated to only query data within the allowed time range. 0 to disable.")
+	f.Var(&l.MaxQueryLookback, "querier.max-query-lookback", "Limit how long back data (series and metadata) can be queried, up until <lookback> duration ago. This limit is enforced in the query-frontend, querier and ruler for instant, range, series metadata and remote read queries. For label name and value queries the limit is enforced in the querier and ruler. If the requested time range is outside the allowed range, the request will not fail but will be manipulated to only query data within the allowed time range. 0 to disable.")
```

Comment:
>This limit is enforced in the query-frontend, querier and ruler for instant, range, series metadata and remote read queries

This limit is enforced in the `limitsMiddleware` in the query-frontend, which doesn't apply to "series metadata" requests. Am I missing anything?

Reply:
i was under the impression that /api/v1/series would go through middlewares, but no it will end up calling Select with the appropriate hints in [series](https://github.com/grafana/mimir/blob/491685e6333b0964742f0d16e1ed9ff225a33e31/vendor/github.com/prometheus/prometheus/web/api/v1/api.go#L873)

Which goes into the querier [Select](https://github.com/grafana/mimir/blob/491685e6333b0964742f0d16e1ed9ff225a33e31/pkg/querier/querier.go#L336)

Which calls the limit checks, so series endpoint is also checked in querier. Updating and merging